realm/realm-library/src/androidTestObjectServer/java/io/realm/SyncUserTests.java

```diff
@@ -356,4 +358,51 @@ public void changePassword_noneAdminThrows() {
         thrown.expect(IllegalStateException.class);
         user.changePassword("user-id", "new-password");
     }
+
+    @Test
+    public void allSessions() {
+        String url1 = "realm://objectserver.realm.io/default";
+        String url2 = "realm://objectserver.realm.io/~/default";
+
+        SyncUser user = createTestUser();
+        assertEquals(0, user.allSessions().size());
+
+        SyncConfiguration configuration1 = new SyncConfiguration.Builder(user, url1).build();
+        Realm realm1 = Realm.getInstance(configuration1);
+        assertEquals(1, user.allSessions().size());
+
+        List<SyncSession> allSessions = user.allSessions();
+        Iterator<SyncSession> iter = allSessions.iterator();
+        SyncSession session = iter.next();
+        assertEquals(user, session.getUser());
+        assertEquals(url1, session.getServerUrl().toString());
+
+        SyncConfiguration configuration2 = new SyncConfiguration.Builder(user, url2).build();
+        Realm realm2 = Realm.getInstance(configuration2);
+        assertEquals(2, user.allSessions().size());
+
+        allSessions = user.allSessions();
+        iter = allSessions.iterator();
+        String indivisualUrl = url2.replace("~", user.getIdentity());
+        int foundCount = 0;
+        while (iter.hasNext()) {
+            session = iter.next();
+            assertEquals(user, session.getUser());
+            if (indivisualUrl.equals(session.getServerUrl().toString())) {
+                foundCount++;
+            }
+        }
+        assertEquals(1, foundCount);
+        realm1.close();
+        assertEquals(1, user.allSessions().size());
```

Comment:
maybe store `allSessions = user.allSessions();` here ,so you can assert the sie then after access the iterator